src/NuGetPackageVerifier/CompositeRules/PackageDependenciesCompositeRule.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGetPackageVerifier.Rules
+{
+    public class PackageDependenciesCompositeRule : IPackageVerifierRule
```

Comment:
Actually, I wonder if the upper-bound/lower-bound thing should be a separate rule that runs per-repo. And then the most important rule about rel/pre-rel deps would be the only one in this new composite rule.